lib/libvarnishapi/vsl_dispatch.c

```diff
@@ -1274,15 +1274,54 @@ vslq_shmref_check(struct VSLQ *vslq)
 	return (vsl_more);
 }
 
+static unsigned
+vslq_candidate(struct VSLQ *vslq, const uint32_t *ptr)
+{
+	enum VSL_transaction_e type;
+	enum VSL_reason_e reason;
+	struct VSL_data *vsl;
+	enum VSL_tag_e tag;
+	unsigned p_vxid;
+	int i;
+
+	CHECK_OBJ_NOTNULL(vslq, VSLQ_MAGIC);
+	AN(ptr);
+
+	assert(vslq->grouping != VSL_g_raw);
+	if (vslq->grouping != VSL_g_vxid)
```

Comment:
I think the commit message could point out that the change only affects vxid grouping.

Reply:
It does, but maybe poorly?

> We also need to collect irrelevant transactions for the sake of
grouping, because even when we do not wish to output them we have
to maintain the hierarchy.

Reply:
I think that is a particularly complicated way to say "only affects vxid grouping". but never mind, it's not so important